### PR TITLE
Fix #204.  Add a program selection to enrollment.

### LINKED ISSUE #204: Enlace - Add program select for program enrollment report
Like assessments and attendance, a set of checkboxes with hide/show for all programs available to the current user, that is then selected by.

### PR DESCRIPTION
**Pull request by frankduncan**
@cecilia-donnelly for your review.

This adds a popup box to the program enrollment report that allows you to select programs.  In order to get any kind of results, you need to select December 2015 for the month/year (though that will be fixed with #206).

### CODE REVIEW

**Comment by frankduncan** [comment]
I admittedly copied and modified the code from the assessments report that also hides what programs you selected, though I now note that the attendance report displays what items you selected.  Right now, the heading of the program report says "Results until" and then the month/year.  Do you think it would be a good interface to have, when programs are selected, it then say "for the following programs:" and then listing them out?  It may become unwieldy if someone selects many programs, but I think that's acceptable because more information is better?

File: enlace/reports/num_programs.php [hunk 1]
```diff
@@ -284,7 +372,7 @@
             <td class="all_projects">
                 <?php
                 //get distinct participants, then count rows
-                $distinct_people = "SELECT DISTINCT Participant_ID FROM Participants_Programs  " . $participant_program_string . " AND MONTH(Date_Added)<='" . $month_select_sqlsafe . "' AND YEAR(Date_Added)<='" . $year_select_sqlsafe . "' AND Date_Dropped IS NOT NULL;";
+                $distinct_people = "SELECT DISTINCT Participant_ID FROM Participants_Programs  " . $participant_program_string . $session_querystring . " AND MONTH(Date_Added)<='" . $month_select_sqlsafe . $session_querystring . "' AND YEAR(Date_Added)<='" . $year_select_sqlsafe . "' AND Date_Dropped IS NOT NULL;";
```
[the remark this answers is not in the document]
Comment (cecilia-donnelly) [reply enlace/reports/num_programs.php]: :) np!

File: enlace/reports/num_programs.php [hunk 2]
```diff
@@ -67,6 +98,45 @@
         <option <?php echo($_POST['year_select'] == 2014 ? 'selected=="selected"' : null) ?>>2014</option>
         <option <?php echo($_POST['year_select'] == 2015 ? 'selected=="selected"' : null) ?>>2015</option>
     </select>
+    <div class="popupcontainer popupcontainer-hidden" style="z-index:2; display:inline; position:relative; text-align:left">
+      <button class="popupdisplay">Select Programs</button>
+      <div class="popupinner" style="width:500px; height:400px; ;white-space:nowrap; overflow:auto;">
+        <div style="position:sticky; width:480px; top:0px; left:0px; padding:4px;  background-color:white; padding-bottom:3; border-bottom:1px solid black ">
```
[the remark this answers is not in the document]
Comment (cecilia-donnelly) [reply enlace/reports/num_programs.php]: That reasoning makes sense!  I just tend to wince when I see this much information held in the `style`.  Thanks, @frankduncan.

**Comment by cecilia-donnelly** [comment]
> Do you think it would be a good interface to have, when programs are selected, it then say "for the following programs:" and then listing them out? It may become unwieldy if someone selects many programs, but I think that's acceptable because more information is better?

Yes, I think that's fine!  It's true that if they select all the list will be quite long, but I suspect they'll more often choose the 10 or so sessions that are taking place currently.

**Comment by frankduncan** [comment]
@cecilia-donnelly Please take a look.  I went a different direction than I had intended and added a row with 0 results for each session checked that didn't have any results.  I first attempted to do this naturally through the query but that proved to be too onerous with the way the query was generated through the file.

**Pull request merged**